manager/controlapi/service.go

```diff
@@ -215,6 +215,10 @@ func (s *Server) checkPortConflicts(spec *api.ServiceSpec) error {
 	}
 
 	for _, service := range services {
+		// If service ID is the same (and not "") then this is an update
+		if serviceID != "" && serviceID == service.ID {
+			continue
```

Comment:
Never mind - misunderstood this. We're just skipping the comparison of this service to itself, but still want to check it against the other services.